WolvenKit.Core/FNV1A/FNV1A64HashAlgorithm.cs

```diff
@@ -3,174 +3,211 @@
-            var hash = FnvHashInitial;
-            foreach (var b in source)
+            unchecked
             {
-                unchecked
-                {
-                    hash = (hash ^ (ulong)b) * FnvHashPrime;
-                }
+                hash = (hash ^ (ulong)b) * FnvHashPrime;
             }
-
-            return hash;
         }
 
-        public static ulong HashString(string value) => HashString(value, Encoding.ASCII, false, false);
+        return hash;
+    }
+
+    public static ulong HashString(string value) => HashString(value, Encoding.ASCII);
+
+    public static ulong HashString(string value, Encoding encoding, bool nullEnded = false, bool useSignedBuffer = false) => 
+        HashString(value.AsSpan(), encoding, nullEnded, useSignedBuffer);
 
-        public static ulong HashString(string value, Encoding encoding, bool nullEnded = false, bool useSignedBuffer = false)
+    public static ulong HashString(ReadOnlySpan<char> value, Encoding encoding, bool nullEnded = false, bool useSignedBuffer = false)
+    {
+        var length = encoding.GetMaxByteCount(nullEnded ? value.Length + 1 : value.Length);
+        var buffer = ArrayPool<byte>.Shared.Rent(length);
+        try
         {
-            var length = encoding.GetMaxByteCount(nullEnded ? value.Length + 1 : value.Length);
-            var buffer = ArrayPool<byte>.Shared.Rent(length);
-            try
-            {
-                var encodedLength = encoding.GetBytes(value.AsSpan(), buffer.AsSpan());
+            var encodedLength = encoding.GetBytes(value, buffer.AsSpan());
 
-                if (nullEnded)
-                {
-                    buffer[encodedLength] = 0;
-                    encodedLength++;
-                }
+            if (nullEnded)
+            {
+                buffer[encodedLength] = 0;
+                encodedLength++;
+            }
 
-                if (useSignedBuffer)
-                {
-                    var signedBuffer = (sbyte[])(Array)buffer;
-                    return HashReadOnlySpan(new ReadOnlySpan<sbyte>(signedBuffer, 0, encodedLength));
-                }
-                else
-                {
-                    return HashReadOnlySpan(new ReadOnlySpan<byte>(buffer, 0, encodedLength));
-                }
+            if (useSignedBuffer)
+            {
+                var signedBuffer = (sbyte[])(Array)buffer;
+                return HashReadOnlySpan(new ReadOnlySpan<sbyte>(signedBuffer, 0, encodedLength));
             }
-            finally
+            else
             {
-                ArrayPool<byte>.Shared.Return(buffer);
+                return HashReadOnlySpan(new ReadOnlySpan<byte>(buffer, 0, encodedLength));
             }
         }
+        finally
+        {
+            ArrayPool<byte>.Shared.Return(buffer);
+        }
+    }
 
-        public void AppendString(string value) => AppendString(value, false);
+    public static ulong HashStringWithoutAliases(ReadOnlySpan<char> value)
```

Comment:
Don't know how others see it, but I'm not a fan of mixing custom hash implementations into core hash classes

Reply:
Well, I agree. This does look out of place. But where else do we put it? In the `NodeRef` itself? No, that breaks "single responsibility" principle. So we have another method to hash an ASCII string.